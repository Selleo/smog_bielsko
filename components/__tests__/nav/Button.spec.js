import 'react-native';
import React                          from 'react';

import { shallow }                    from 'enzyme';
import chai, { expect }               from 'chai';
import dirtyChai                      from 'dirty-chai';

import Button                         from '../../nav/Button';

const originalExpect = global.expect;

chai.use(dirtyChai);

let wrapper, props, closeDrawerSpy;

describe('Button', () => {
  beforeEach(() => {
    props = {
      style: 45, //nr because its type of styles
      onPress: (onPress) => closeDrawerSpy = 2
    };

    wrapper = shallow(<Button {...props}/>);
  });

  it('renders correctly', () => {
    originalExpect(wrapper).toMatchSnapshot();
    expect(wrapper).to.exist();
  });

  describe('onPress', () => {
    it('change closeDrawer', () => {
      expect(closeDrawerSpy).to.be.an('undefined');
      wrapper.find('TouchableOpacity').simulate('press');
      expect(closeDrawerSpy).to.equal(2);
    });
  });
});
