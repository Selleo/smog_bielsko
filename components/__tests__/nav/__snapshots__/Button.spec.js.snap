exports[`Button renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    onPress={[Function]}
    style={45}>
    <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail" />
</TouchableOpacity>,
  "nodes": Array [
    <TouchableOpacity
      activeOpacity={0.2}
      focusedOpacity={0.7}
      onPress={[Function]}
      style={45}>
      <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail" />
</TouchableOpacity>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Button
        onPress={[Function]}
        style={45} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Button {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "onPress": [Function],
          "style": 45,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <TouchableOpacity
          activeOpacity={0.2}
          focusedOpacity={0.7}
          onPress={[Function]}
          style={45}>
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail" />
</TouchableOpacity>,
        "_debugID": 2,
        "_renderedOutput": <TouchableOpacity
          activeOpacity={0.2}
          focusedOpacity={0.7}
          onPress={[Function]}
          style={45}>
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail" />
</TouchableOpacity>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Button
    onPress={[Function]}
    style={45} />,
}
`;
